### PR TITLE
ENT-11970: Moved masterfiles-stage checks for git in sync so that they can work properly

### PR DESCRIPTION
**Pull request by craigcomstock**
Ticket: ENT-11970
Changelog: none

### CODE REVIEW

File: contrib/masterfiles-stage/common.sh [hunk 1]
```diff
@@ -385,15 +380,22 @@ git_masterstage() {
   # Depends on $GIT_URL, $ROOT, $MASTERDIR, $GIT_REFSPEC
   check_git_installed
   git_setup_local_mirrored_repo "$( dirname "$ROOT" )"
-  if [ "x$1" = "xtrue" ]; then
-    if git_check_is_in_sync "$( dirname "$ROOT" )" "$MASTERDIR" "$GIT_REFSPEC"; then
-      return 1  # in sync => nothing to do
+  git_check_is_in_sync "$local_mirrored_repo" "$MASTERDIR" "$GIT_REFSPEC"
```
Comment (craigcomstock) [inline contrib/masterfiles-stage/common.sh:383]: Might be nice to mention in a comment here that we are abusing the previous function setting a global variable so we know local_mirrored_repo

File: contrib/masterfiles-stage/common.sh [hunk 2]
```diff
@@ -59,6 +59,7 @@ git_setup_local_mirrored_repo() {
   # /opt/cfengine/git_mygitserver.net_joeblow_my_policy_repo.git
 
   if [ -d "${local_mirrored_repo}" ] ; then
+    git --git-dir="${local_mirrored_repo}" fetch --dry-run # to provide information in logs about from and to commits
```
Comment (vpodzime) [inline contrib/masterfiles-stage/common.sh:62]: How does this make it logged more than the run without `--dry-run` below?

**Review by vpodzime — COMMENTED** [review]
I don't quite understand what's going on here. Please explain things in the commit messages.

File: contrib/masterfiles-stage/common.sh (hunk 2 above)
Comment (craigcomstock) [reply contrib/masterfiles-stage/common.sh:62]: I suppose that's true. I wasn't seeing anything without this.

File: contrib/masterfiles-stage/common.sh [hunk 3]
```diff
@@ -112,9 +112,6 @@ git_deploy_refspec() {
   mkdir -p "${temp_stage}/.git"
   cp "${local_mirrored_repo}/HEAD" "${temp_stage}/.git/"
 
-  if git_check_is_in_sync "${local_mirrored_repo}" "${temp_stage}" "$2"; then
-    return 0
-  fi
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply contrib/masterfiles-stage/common.sh:117]: The problem here I think is that we have copied HEAD into temp_stage so it is always the same.

As-is I never got updates. There may be another cause but if you check my notes in ENT-11970 you might see some details that shed some light on the situation.

I imagine it worked for you while testing so not sure what is going on. Maybe grab a nightly build and try it out there. That's what I was doing.

[the remark this answers is not in the document]
Comment (vpodzime) [reply contrib/masterfiles-stage/common.sh:117]: > I imagine it worked for you while testing so not sure what is going on.

Unfortunately, I didn't test this very much. I was planning to test the whole set of changing how and where from *masterfiles-stage.sh* is run, but then I got busy with other things before finishing the set.

[the remark this answers is not in the document]
Comment (vpodzime) [reply contrib/masterfiles-stage/common.sh:117]: > The problem here I think is that we have copied HEAD into temp_stage so it is always the same.

Oh, I see it now! But IMHO the error here is to compare with `${temp_stage}`, we should compare with `$MASTERDIR` (or whatever it's called) and keep doing it here.

[the remark this answers is not in the document]
Comment (vpodzime) [reply contrib/masterfiles-stage/common.sh:117]: yup, looks like that's enough: https://github.com/cfengine/core/pull/5579 (and https://github.com/cfengine/nova/pull/2284)

**Comment by craigcomstock** [comment]
Other fixes have been made. No need for this PR any more. :)

**Pull request closed without merging**